### DISCUSSION TITLE
Dashboard assets

### DISCUSSION THREAD

**Discussion by Alexplose** [elsa-workflows/elsa-core#560] [Q&A]
Sometime dashboard _content assets are not found, sometimes they are, and sometimes I have to manually copy the directory in wwwroot.

What is the good practice about theses files ? how to make sure they are found in the nugget directory ?

.nuget\packages\elsa.dashboard\1.4.5.1\staticwebassets

**Comment by sfmskywalker** [comment]
That's odd. They should always be available from the Nuget package, so I'm surprised to hear that sometimes they are present and other times they aren't. By "sometimes", do you mean "on some HTTP requests", or "on certain versions of the package"?

**Reply by Alexplose** [reply]
I'm not clear, they are always present in the nugget package, but when I run my web app and open the dashboard the assets files are all 404, I have to copy them manually in wwwroot in the right directory : wwwroot\_content\Elsa.Dashboard\assets

And sometimes when I build I get an error telling me directories are conflicting.

I don't have the error right now I'll add the details when I get it.

**Reply by Alexplose** [reply]
Here is the error  : 
`Error		The static web asset 'C:\Users\Alex\.nuget\packages\elsa.dashboard\1.4.5.1\build\..\staticwebassets\assets\css\argon-dashboard.css' has a conflicting web root path '/wwwroot/_content/Elsa.Dashboard/assets/css/argon-dashboard.css' with the project file 'wwwroot\_content\Elsa.Dashboard\assets\css\argon-dashboard.css'.	PGP	C:\Program Files\dotnet\sdk\5.0.101\Sdks\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Microsoft.NET.Sdk.Razor.StaticWebAssets.targets	208`

**Reply by SachinMehndiratta** [reply]
This issue is reproducible to me atleast.

I even tried to launch the Elsa2 Aspnetcore monolith dashboard sample and encountered the same

![image](https://user-images.githubusercontent.com/29795134/117499972-a2231f80-af99-11eb-97b6-fe4cc69effe6.png)

**Reply by SachinMehndiratta** [reply]
> Here is the error :
> `Error The static web asset 'C:\Users\Alex\.nuget\packages\elsa.dashboard\1.4.5.1\build\..\staticwebassets\assets\css\argon-dashboard.css' has a conflicting web root path '/wwwroot/_content/Elsa.Dashboard/assets/css/argon-dashboard.css' with the project file 'wwwroot\_content\Elsa.Dashboard\assets\css\argon-dashboard.css'. PGP C:\Program Files\dotnet\sdk\5.0.101\Sdks\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Microsoft.NET.Sdk.Razor.StaticWebAssets.targets 208`

@Alexplose - By any chance, do you got any solution ?

**Reply by Alexplose** [reply]
Still no solution to this issue, still have it sometimes, will try updating to latest version later to see if it is still an issue

**Reply by sfmskywalker** [reply]
For Elsa 2, make sure to build the assets of the designer project as well as the bindings project. I’ll document this for future reference.

**Reply by tunechr** [reply]
I ge the same 404 when running the Monolithic. 

I also get the following error when trying to install the designer package. 

Error	TS18003	No inputs were found in config file 'C:/Users/xxx/.nuget/packages/elsa.designer.components.web/2.0.0-rc1.31/contentFiles/any/net5.0/tsconfig.json'. Specified 'include' paths were '["**/*"]' and 'exclude' paths were '[]'.	C:\Users\xxx\.nuget\packages\elsa.designer.components.web\2.0.0-rc1.31\contentFiles\any\net5.0\tsconfig.json		1	Active

**Reply by SachinMehndiratta** [reply]
Deleting the package.json and tsconfig.json from project get rid of the error and everything starts working fine for me

**Reply by sfmskywalker** [reply]
I'll remove all the typescript configs from the bindings project and see if that helps. We're not using any of that anyway. Before I push the packages to NuGet, can I ask you to try the MyGet packages first to see if the issue is resolved?

Here's how to update to the latest packages from the MyGet feed: http://localhost:4000/elsa-core/docs/next/installation/installing-feeds#myget, which should be up there once [this build](https://ci.appveyor.com/project/sfmskywalker/elsa/builds/39084942) completes.

**Reply by tunechr** [reply]
Thanks That worked!

BTW the link in your message is linking to your localhost 
https://elsa-workflows.github.io/elsa-core/docs/next/installation/installing-feeds

**Reply by sfmskywalker** [reply]
Indeed. Happy to see you found the correct one 😅

**Reply by sfmskywalker** [reply]
I also updated the README on the project homepage with instructions to build the client assets for future reference.
